src/event-log/event-log-level.ts

```diff
@@ -15,101 +18,114 @@ type EventLogLevelConfig = {
   createLevelDatabase?: typeof createLevelDatabase,
 };
 
-const WATERMARKS_SUBLEVEL_NAME = 'watermarks';
 const CIDS_SUBLEVEL_NAME = 'cids';
 
 export class EventLogLevel implements EventLog {
-  config: EventLogLevelConfig;
-  db: LevelWrapper<string>;
   ulidFactory: ULIDFactory;
+  index: IndexLevel<Event>;
 
   constructor(config?: EventLogLevelConfig) {
-    this.config = {
+    this.index = new IndexLevel({
       location: 'EVENTLOG',
       createLevelDatabase,
       ...config,
-    };
-
-    this.db = new LevelWrapper<string>({
-      location            : this.config.location!,
-      createLevelDatabase : this.config.createLevelDatabase,
-      valueEncoding       : 'utf8',
     });
+
     this.ulidFactory = monotonicFactory();
   }
 
   async open(): Promise<void> {
-    return this.db.open();
+    return this.index.open();
   }
 
   async close(): Promise<void> {
-    return this.db.close();
+    return this.index.close();
   }
 
   async clear(): Promise<void> {
-    return this.db.clear();
+    return this.index.clear();
   }
 
-  async append(tenant: string, messageCid: string): Promise<string> {
-    const tenantEventLog = await this.db.partition(tenant);
-    const watermarkLog = await tenantEventLog.partition(WATERMARKS_SUBLEVEL_NAME);
-    const cidLog = await tenantEventLog.partition(CIDS_SUBLEVEL_NAME);
-
+  /**
+   * Appends Events to the EventLog in the order they are appended using a ulid watermark.
+   * The indexes are used to query the event-log for specific events.
+   *
+   * @param tenant
+   * @param messageCid the messageCid of the message that is being appended to the log.
+   * @param indexes property and value indexes to use for querying potential events.
+   * @returns the ulid watermark generated during this operation.
+   */
```

Comment:
Remove these comments as well as for methods below) or move to/consolidate with `EventLog`. Since you are implementing `EventLog`, there is no need to duplicate the documentation, code auto-completion will show the documentation from the interface directly, as it should be.